drivers/sensor/mpu9250/mpu9250.c

```diff
@@ -0,0 +1,299 @@
+				 uint16_t sensitivity_x10)
+{
+	int64_t conv_val;
+
+	conv_val = ((int64_t)raw_val * SENSOR_PI * 10) /
+		   (sensitivity_x10 * 180U);
+	val->val1 = conv_val / 1000000;
+	val->val2 = conv_val % 1000000;
+}
+
+/* see "Temperature Measurement" section from register map description */
+static inline void mpu9250_convert_temp(struct sensor_value *val,
+					int16_t raw_val)
+{
+	val->val1 = raw_val / 340 + 36;
+	val->val2 = ((int64_t)(raw_val % 340) * 1000000) / 340 + 530000;
+
+	if (val->val2 < 0) {
+		val->val1--;
+		val->val2 += 1000000;
+	} else if (val->val2 >= 1000000) {
+		val->val1++;
+		val->val2 -= 1000000;
+	}
+}
+
+static int mpu9250_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct mpu9250_data *drv_data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_XYZ:
+		mpu9250_convert_accel(val, drv_data->accel_x,
+				      drv_data->accel_sensitivity_shift);
+		mpu9250_convert_accel(val + 1, drv_data->accel_y,
+				      drv_data->accel_sensitivity_shift);
+		mpu9250_convert_accel(val + 2, drv_data->accel_z,
+				      drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_ACCEL_X:
+		mpu9250_convert_accel(val, drv_data->accel_x,
+				      drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		mpu9250_convert_accel(val, drv_data->accel_y,
+				      drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		mpu9250_convert_accel(val, drv_data->accel_z,
+				      drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_GYRO_XYZ:
+		mpu9250_convert_gyro(val, drv_data->gyro_x,
+				     drv_data->gyro_sensitivity_x10);
+		mpu9250_convert_gyro(val + 1, drv_data->gyro_y,
+				     drv_data->gyro_sensitivity_x10);
+		mpu9250_convert_gyro(val + 2, drv_data->gyro_z,
+				     drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_X:
+		mpu9250_convert_gyro(val, drv_data->gyro_x,
+				     drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_Y:
+		mpu9250_convert_gyro(val, drv_data->gyro_y,
+				     drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_Z:
+		mpu9250_convert_gyro(val, drv_data->gyro_z,
+				     drv_data->gyro_sensitivity_x10);
+#ifdef CONFIG_MPU9250_MAGN_EN
+	case SENSOR_CHAN_MAGN_XYZ:
+		ak8963_convert_magn(val, drv_data->magn_x,
+				    drv_data->magn_scale_x);
+		ak8963_convert_magn(val + 1, drv_data->magn_y,
+				    drv_data->magn_scale_y);
+		ak8963_convert_magn(val + 2, drv_data->magn_z,
+				    drv_data->magn_scale_z);
+		break;
+	case SENSOR_CHAN_MAGN_X:
+		ak8963_convert_magn(val, drv_data->magn_x,
+				    drv_data->magn_scale_x);
+		break;
+	case SENSOR_CHAN_MAGN_Y:
+		ak8963_convert_magn(val, drv_data->magn_y,
+				    drv_data->magn_scale_y);
+		break;
+	case SENSOR_CHAN_MAGN_Z:
+		ak8963_convert_magn(val, drv_data->magn_z,
+				    drv_data->magn_scale_z);
+#endif
+		break;
+	default: /* chan == SENSOR_CHAN_DIE_TEMP */
```

Comment:
return `-ENOTSUP` for unsupported channels, not `0`

Reply:
Fixed.